hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -131,7 +131,7 @@ public void destroy() {
 
     @Override
     public void destroyInternals() {
-        clearIndexes();
+        clearPartitionedIndexes();
```

Comment:
Probably it worth adding a comment why we are clearing only partitioned indexes here and where the global ones are cleared.